src/watertap_contrib/seto/unit_models/surrogate/vagmd_surrogate.py

```diff
@@ -0,0 +1,1309 @@
+        ),
+    )
+    CONFIG.declare(
+        "property_package_seawater",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for control volume",
+            doc="""Property parameter object used to define property calculations,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_water",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for control volume",
+            doc="""Property parameter object used to define property calculations,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigBlock with arguments to be passed to a property block(s)
+    and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+    CONFIG.declare(
+        "high_brine_salinity",
+        ConfigValue(
+            default=False,
+            domain=Bool,
+            doc="""A ConfigBlock with arguments to be passed to indicate if the brine
+            has a salinity higher than 175.3 g/L""",
+        ),
+    )
+    CONFIG.declare(
+        "module_type",
+        ConfigValue(
+            default="AS7C1.5L",
+            domain=In(["AS7C1.5L", "AS26C7.2L"]),
+            doc="""Selection of module type (7 for AS7C1.5L and 26 for AS26C7.2L)""",
+        ),
+    )
+    CONFIG.declare(
+        "cooling_system_type",
+        ConfigValue(
+            default="closed",
+            domain=In(["open", "closed"]),
+            doc="""Selection of cooling system type (open or closed)""",
+        ),
+    )
+
+    def build(self):
+        super().build()
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        units_meta = (
+            self.config.property_package_seawater.get_metadata().get_derived_units
+        )
+
+        """
+        Model parameters
+        """
+        self.gas_constant_1 = Param(
+            initialize=8.314,
+            units=pyunits.J / pyunits.K / pyunits.mol,
+            doc="Gas constan in J/K mol",
+        )
+
+        self.gas_constant_2 = Param(
+            initialize=0.082,
+            units=pyunits.atm * pyunits.L / pyunits.K / pyunits.mol,
+            doc="Gas constant in atm * L /K mol",
+        )
```

Comment:
You can import the ideal gas constant from IDAES and convert units on that constant as needed and delete these

Reply:
Sure, will incorporate it in the new PR.